### ISSUE TITLE
classes inheriting from typing.Protocol with @property decorators are interpreted differently between mypy and pyright

### ISSUE THREAD

**Issue by Thirdegree** [microsoft/pyright#2490]
**Describe the bug**
classes inheriting from typing.Protocol with @property decorators are interpreted differently between mypy and pyright

**To Reproduce**
```python
#python3.9, also applicable in 3.6 from typing_extensions
from typing import Protocol

class MyProtocol(Protocol):
    @property
    def my_int(self) -> int: ...
    @property
    def my_str(self) -> str: ...

def my_func(prot: MyProtocol) -> None:
    pass

class Whatever:
    def __init__(self) -> None:
        self.my_int = 5
        self.my_str = 'foo'

if __name__ == '__main__':
    my_func(Whatever())
```

Then `pyright test.py`

**Expected behavior**
No errors. 

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```
Found 1 source file
/home/thirdegree/.virtualenvs/tmp-fc1289085245298b/test.py
  /home/thirdegree/.virtualenvs/tmp-fc1289085245298b/test.py:19:13 - error: Argument of type "Whatever" cannot be assigned to parameter "prot" of type "MyProtocol" in function "my_func"
    "Whatever" is incompatible with protocol "MyProtocol"
      "my_int" is an incompatible type
        "int" is incompatible with "property"
      "my_str" is an incompatible type
        "str" is incompatible with "property" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
Completed in 0.679sec
```

**VS Code extension or command-line**
```
$ pyright --version
pyright 1.1.177
```
**Additional context**
